aiida/engine/processes/functions.py

```diff
@@ -15,23 +15,49 @@
 import inspect
 import logging
 import signal
-from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, Sequence, Tuple, Type, TypeVar
+import types
+import typing as t
+from typing import TYPE_CHECKING
 
 from aiida.common.lang import override
 from aiida.manage import get_manager
-from aiida.orm import CalcFunctionNode, Data, ProcessNode, WorkFunctionNode, to_aiida_type
+from aiida.orm import (
+    Bool,
+    CalcFunctionNode,
+    Data,
+    Dict,
+    Float,
+    Int,
+    List,
+    ProcessNode,
+    Str,
+    WorkFunctionNode,
+    to_aiida_type,
+)
 from aiida.orm.utils.mixins import FunctionCalculationMixin
 
 from .process import Process
 
+try:
+    UnionType = types.UnionType
+except AttributeError:
+    # This type is not available for Python 3.9 and older
+    UnionType = None  # pylint: disable=invalid-name
+
+try:
+    get_annotations = inspect.get_annotations
+except AttributeError:
+    # This is the backport for Python 3.9 and older
+    from get_annotations import get_annotations
+
 if TYPE_CHECKING:
     from .exit_code import ExitCode
 
 __all__ = ('calcfunction', 'workfunction', 'FunctionProcess')
 
 LOGGER = logging.getLogger(__name__)
 
-FunctionType = TypeVar('FunctionType', bound=Callable[..., Any])
+FunctionType = t.TypeVar('FunctionType', bound=t.Callable[..., t.Any])
```

Comment:
Note in #5901 I have started naming these with a `Tv` suffix, like `FunctionTv`, because pylint no longer allows `Type` https://github.com/PyCQA/pylint/issues/6003

Also can you make sure this file is no longer in the exclude list in the pre-commit-config

Reply:
May I ask the reason for choosing `Tv` as the suffix? When going over #5901 it already struck me a bit weird, since I read it automatically as "television" 😅  I wasn't going to mention it, since in the end it _is_ an arbitrary choice, but if it is `pylint` that complains, why not use [what they suggest as naming](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/invalid-name.html#predefined-naming-patterns) namely `T` as suffix. This also makes more sense to me. I would propose we just maintain that convention. What do you think?

Reply:
@chrisjsewell I have removed the file from the black list. Since the naming of the type var is still  an open discussion and the latest version of pylint is anyway not being used, I will leave that for another PR. Since you had no other comments, I take it this is good to go and will merge tomorrow.